packages/core/src/redocly/index.ts

```diff
@@ -1,78 +1,106 @@
 import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
 import { resolve } from 'path';
 import { homedir } from 'os';
-import { yellow, red, green, gray } from 'colorette';
+import { red, green, gray } from 'colorette';
 import { RegistryApi } from './registry-api';
+import { DEFAULT_DOMAIN, DEFAULT_REGION, DOMAINS, Region } from '../config/config';
 
 const TOKEN_FILENAME = '.redocly-config.json';
 
 export class RedoclyClient {
-  private accessToken: string | undefined;
+  private accessTokens: { us?: string; eu?: string; } | undefined;
+  private region: Region;
+  domain: string;
   registryApi: RegistryApi;
 
-  constructor() {
-    this.loadToken();
-    this.registryApi = new RegistryApi(this.accessToken);
+  constructor(region?: Region) {
+    this.region = this.loadRegion(region);
+    this.loadTokens();
+    this.domain = region
+      ? DOMAINS[region]
+      : process.env.REDOCLY_DOMAIN || DEFAULT_DOMAIN;
+    this.registryApi = new RegistryApi(this.getTokenByRegion(), this.domain);
   }
 
-  hasToken(): boolean {
-    return !!this.accessToken;
+  loadRegion(region?: Region) {
+    if (region && !DOMAINS[region]) {
+      process.stdout.write(
+        red(`Invalid argument: region in config file.\nGiven: ${green(region)}, choices: "us", "eu".\n`),
+      );
+      process.exit(1);
+    }
+    return region || DEFAULT_REGION;
   }
 
-  loadToken(): void {
-    if (process.env.REDOCLY_AUTHORIZATION) {
-      this.accessToken = process.env.REDOCLY_AUTHORIZATION;
-      return;
-    }
+  getTokenByRegion() {
+    return this.accessTokens && this.accessTokens[this.region];
+  }
 
+  hasTokens(): boolean {
+    return !!this.accessTokens;
+  }
+
+  loadTokens(): void {
```

Comment:
add unit test for this method

Reply:
@andriyl why did you resolve this with no comments? and no tests? 

add tests like this 
```
it('should loadTokens with no credentials ( NO .redocly-config.json)', async () => {
    const client = new RedoclyClient();
    const data = {};
    jest.spyOn(RedoclyClient.prototype, 'setAccessTokens');
    let spy = jest.spyOn(RedoclyClient.prototype, 'readCredentialsFile').mockImplementation(() => data);

    await client.loadTokens();
    expect(client.setAccessTokens).toHaveBeenCalledTimes(0);
    spy.mockRestore();
  });

  it('should loadTokens with old format credentials', async () => {
    const client = new RedoclyClient();
    const data = {"token": "test"};
    const expected = {"token": "test", "us": "test"};
    jest.spyOn(RedoclyClient.prototype, 'setAccessTokens');
    let spy = jest.spyOn(RedoclyClient.prototype, 'readCredentialsFile').mockImplementation(() => data);

    await client.loadTokens();
    expect(client.setAccessTokens).toHaveBeenLastCalledWith(expected);
    spy.mockRestore();
  });
  // it('should loadTokens with combination of old and new format', async () => {
    //   const data = {
    //     "token": "token old",
    //     "us": "token new"
    //   };
  // });
  // it('should loadTokens with "US"" region only', async () => {
  // });
  // it('should loadTokens with "EU" region only', async () => {
  // });
  // it('should loadTokens with process.env.REDOCLY_AUTHORIZATION + credentials', async () => {
  // });
  // it('should loadTokens with process.env.REDOCLY_AUTHORIZATION - credentials', async () => {
  // });
  ```
  
  maybe there are more use cases...

Reply:
done